app/src/main/java/org/schabi/newpipe/player/BasePlayer.java

```diff
@@ -497,12 +497,19 @@ public void onLoadingFailed(final String imageUri, final View view,
     @Override
     public void onLoadingComplete(final String imageUri, final View view,
                                   final Bitmap loadedImage) {
+        final float width = Math.min(
+                context.getResources().getDimension(R.dimen.player_notification_thumbnail_width),
+                loadedImage.getWidth());
+        currentThumbnail = Bitmap.createScaledBitmap(loadedImage,
+                (int) width,
+                (int) (loadedImage.getHeight() / (loadedImage.getWidth() / width)), true);
         if (DEBUG) {
             Log.d(TAG, "Thumbnail - onLoadingComplete() called with: "
                     + "imageUri = [" + imageUri + "], view = [" + view + "], "
-                    + "loadedImage = [" + loadedImage + "]");
+                    + "loadedImage = [" + loadedImage + "], "
+                    + loadedImage.getWidth() + "x" + loadedImage.getHeight()
+            + ", scaled width = " + width);
         }
-        currentThumbnail = loadedImage;
     }
```

Comment:
Is this change needed with the MedidaStyle notifications?

Reply:
I think so, as doing this in the background prevents it from being done by Android on the main thread. Also, I think the slowdown is not related to MediaStyle, but rather to the MediaSession, so this code probably helps independently of the type of notifications used. @vkay94 correct me if I'm wrong ;-)

Reply:
In this place
```java
Bitmap.createScaledBitmap
```
I'm scaling down the image only once before assigning it to `currentThumbnail`. Without it Android will resize the thumbnail on every notification update which is a heavy task if the image is large (on YouTube it's large). In this case I made it smaller (much smaller) and if Android needs to resize it the time needed for the resize will be small. I'm not sure about the quality of the thumbnail but for me it looks ok compared to previous thumbnail impelementation. You can then change the width from 200dp (player_notification_thumbnail_width) to something else if the quality is not as good as you want.

The main thing that eats main thread's CPU time is an image resize but after this change there is no such problem. Actually I didn't tested in profiler after the change but visually it doesn't lock main thread as before on rotation change and after other changes.

Reply:
With media style notifications the thumbnail can be even smaller because it use less space than previous notifications.

Reply:
Would be better to somehow get the size for thumbnail inside media style notification and apply this size instead of 200dp. Does anyone know the exact size Android uses for notification?

Reply:
no. I assume that varies from screen size and density.